assets/model_monitoring/components/model_performance/model_performance_compute_metrics/spec.yaml

```diff
@@ -0,0 +1,78 @@
+$schema: http://azureml/sdk-2-0/SparkComponent.json
+type: spark
+
+name: model_performance_compute_metrics
+display_name:  Model Performance - Compute Metrics
+description: Compute model performance metrics leveraged by the model performance monitor.
+version: 0.0.2
+is_deterministic: true
+
+code: ../../src
+entry:
+  file: ./model_performance_metrics/compute_metrics.py
+
+inputs:
+  task:
+    type: string
+    optional: false
+    default: tabular-classification
+    enum: [
+      tabular-classification,
+      tabular-classification-multilabel,
+      tabular-regression
+    ]
+    description: "Task type"
+  baseline_data:
+    type: mltable
+    mode: direct
+  baseline_data_column_name:
+    type: string
+    optional: true
+    description: "Column name which contains ground truths in provided input for baseline_data. (Optional if we have one column name.)"
+  production_data:
+    type: mltable
+    mode: direct
+  production_data_column_name:
+    type: string
+    optional: true
+    description: "Column name which contains predictions in provided input for production_data_column_name. (Optional if we have one column name.)"
+
+outputs:
+  signal_metrics:
+    type: mltable
+    mode: direct
+
+conf:
+  spark.driver.cores: 1
+  spark.driver.memory: 2g
+  spark.executor.cores: 2
+  spark.executor.memory: 2g
+  spark.executor.instances: 1
+  spark.dynamicAllocation.enabled: True
+  spark.dynamicAllocation.minExecutors: 1
+  spark.dynamicAllocation.maxExecutors: 4
+  spark.synapse.library.python.env: |
+    channels:
+      - conda-forge
+    dependencies:
+      - python=3.8
+      - pip:
+        - scipy~=1.10.0
+        - numpy~=1.21.0
+        - pandas~=1.4.3
+        - azureml-metrics~=0.0.27
+        - mltable~=1.3.0
+        - azureml-fsspec
+        - fsspec~=2023.4.0
+    name: momo-base-spark
+args: >-
+  --task ${{inputs.task}}
+  --baseline_data ${{inputs.baseline_data}}
+  --$[[baseline_data_column_name ${{inputs.baseline_data_column_name}}]]
+  --production_data ${{inputs.production_data}}
+  --$[[production_data_column_name ${{inputs.production_data_column_name}}]]
+  --signal_metrics ${{outputs.signal_metrics}}
+
+resources:
```

Comment:
nit: This is not needed in spark component spec. We usually define this in the parent pipeline component spec.